Index: driverspecs.h
===================================================================
--- driverspecs.h	(revision 91)
+++ driverspecs.h	(working copy)
@@ -278,8 +278,8 @@
 	#define __drv_in_deref(annotes) __pre __deref __$drv_group(##__drv_nop(annotes))
 	#define __drv_out(annotes) __post __$drv_group(##__drv_nop(annotes))
 	#define __drv_out_deref(annotes) __post __deref __$drv_group(##__drv_nop(annotes))
-	#define __drv_when(cond, annotes)										\
-	  __drv_declspec("SAL_when(" SPECSTRINGIZE(cond) ")") __$drv_group(##__drv_nop(annotes))
+	#define __drv_when(cond, annotes)
+	  //__drv_declspec("SAL_when(" SPECSTRINGIZE(cond) ")") __$drv_group(##__drv_nop(annotes))
 	#define __drv_at(expr,annotes)\
 	  __drv_declspec("SAL_at(" SPECSTRINGIZE(expr) ")") __$drv_group(##__drv_nop(annotes))
 
@@ -354,8 +354,8 @@
 	// This can be tested by the condition function inFunctionClass$()
 	//
 	__ANNOTATION(SAL_functionClass(__in char *);)
-	#define __drv_functionClass(x)									     	\
-		__drv_out(__drv_declspec("SAL_functionClass(\""#x"\")"))
+	#define __drv_functionClass(x)
+		//__drv_out(__drv_declspec("SAL_functionClass(\""#x"\")"))
 
 	// ---------------------------------------------------------------------
 	// Resources:
@@ -650,7 +650,7 @@
 	//
 	__ANNOTATION(SAL_error(__in __AuToQuOtE char *);)
 	#define __drv_reportError(why)											\
-		__pre __drv_declspec("SAL_error(" SPECSTRINGIZE(why) ")")
+		//__pre __drv_declspec("SAL_error(" SPECSTRINGIZE(why) ")")
 
 	// ---------------------------------------------------------------------
 	// Floating point save/restore:
Index: ktmtypes.h
===================================================================
--- ktmtypes.h	(revision 91)
+++ ktmtypes.h	(working copy)
@@ -24,7 +24,9 @@
 #endif
 
 
-typedef GUID UOW, *PUOW;
+typedef GUID UOW;
+typedef UOW* PUOW;
+
 typedef GUID CRM_PROTOCOL_ID, *PCRM_PROTOCOL_ID;
 
 //
@@ -139,7 +141,7 @@
 
 typedef struct _TRANSACTION_NOTIFICATION_RECOVERY_ARGUMENT {
     GUID   EnlistmentId;
-    UOW    UOW;
+    ::UOW    UOW;
 } TRANSACTION_NOTIFICATION_RECOVERY_ARGUMENT, *PTRANSACTION_NOTIFICATION_RECOVERY_ARGUMENT;
 
 #define TRANSACTION_NOTIFICATION_TM_ONLINE_FLAG_IS_CLUSTERED 0x1
@@ -184,7 +186,7 @@
 } KCRM_MARSHAL_HEADER, *PKCRM_MARSHAL_HEADER, *RESTRICTED_POINTER PRKCRM_MARSHAL_HEADER;
 
 typedef struct _KCRM_TRANSACTION_BLOB {
-    UOW                UOW;
+    ::UOW                UOW;
     GUID               TmIdentity;
     ULONG              IsolationLevel;
     ULONG              IsolationFlags;
Index: OAIdl.h
===================================================================
--- OAIdl.h	(revision 91)
+++ OAIdl.h	(working copy)
@@ -442,7 +442,7 @@
                 FLOAT fltVal;
                 DOUBLE dblVal;
                 VARIANT_BOOL boolVal;
-                _VARIANT_BOOL bool;
+//                _VARIANT_BOOL bool;
                 SCODE scode;
                 CY cyVal;
                 DATE date;
@@ -457,7 +457,7 @@
                 FLOAT *pfltVal;
                 DOUBLE *pdblVal;
                 VARIANT_BOOL *pboolVal;
-                _VARIANT_BOOL *pbool;
+//                _VARIANT_BOOL *pbool;
                 SCODE *pscode;
                 CY *pcyVal;
                 DATE *pdate;
Index: ObjBase.h
===================================================================
--- ObjBase.h	(revision 91)
+++ ObjBase.h	(working copy)
@@ -236,7 +236,7 @@
 {
     template<typename T> void** IID_PPV_ARGS_Helper(T** pp) 
     {
-        static_cast<IUnknown*>(*pp);    // make sure everyone derives from IUnknown
+        //static_cast<IUnknown*>(*pp);    // make sure everyone derives from IUnknown
         return reinterpret_cast<void**>(pp);
     }
 }
Index: PropIdl.h
===================================================================
--- PropIdl.h	(revision 91)
+++ PropIdl.h	(working copy)
@@ -316,7 +316,7 @@
         FLOAT fltVal;
         DOUBLE dblVal;
         VARIANT_BOOL boolVal;
-        _VARIANT_BOOL bool;
+//        _VARIANT_BOOL bool;
         SCODE scode;
         CY cyVal;
         DATE date;
Index: WinNT.h
===================================================================
--- WinNT.h	(revision 91)
+++ WinNT.h	(working copy)
@@ -799,13 +799,6 @@
     __in DWORD ShiftCount
     )
 {
-    __asm    {
-        mov     ecx, ShiftCount
-        mov     eax, dword ptr [Value]
-        mov     edx, dword ptr [Value+4]
-        shld    edx, eax, cl
-        shl     eax, cl
-    }
 }
 
 __inline LONGLONG
@@ -815,13 +808,6 @@
     __in DWORD ShiftCount
     )
 {
-    __asm {
-        mov     ecx, ShiftCount
-        mov     eax, dword ptr [Value]
-        mov     edx, dword ptr [Value+4]
-        shrd    eax, edx, cl
-        sar     edx, cl
-    }
 }
 
 __inline ULONGLONG
@@ -831,13 +817,6 @@
     __in DWORD ShiftCount
     )
 {
-    __asm    {
-        mov     ecx, ShiftCount
-        mov     eax, dword ptr [Value]
-        mov     edx, dword ptr [Value+4]
-        shrd    eax, edx, cl
-        shr     edx, cl
-    }
 }
 
 #if _MSC_VER >= 1200
@@ -2232,7 +2211,7 @@
 //
 
 C_ASSERT((sizeof(XSAVE_FORMAT) & (XSAVE_ALIGN - 1)) == 0);
-C_ASSERT((FIELD_OFFSET(XSAVE_AREA, Header) & (XSAVE_ALIGN - 1)) == 0);
+//C_ASSERT((FIELD_OFFSET(XSAVE_AREA, Header) & (XSAVE_ALIGN - 1)) == 0);
 
 // XSAVE_AREA structure must be sized uniformly on all architectures
 C_ASSERT(MINIMAL_XSTATE_AREA_LENGTH == 512 + 64);
@@ -3722,12 +3701,6 @@
     __in LONG Bit
     )
 {
-    __asm {
-           mov eax, Bit
-           mov ecx, Base
-           lock bts [ecx], eax
-           setc al
-    };
 }
 
 FORCEINLINE
@@ -3737,12 +3710,6 @@
     __in LONG Bit
     )
 {
-    __asm {
-           mov eax, Bit
-           mov ecx, Base
-           lock btr [ecx], eax
-           setc al
-    };
 }
 #pragma warning(pop)
 
@@ -3793,12 +3760,6 @@
     __in LONG Bit
     )
 {
-    __asm {
-           mov eax, Bit
-           mov ecx, Base
-           lock btc [ecx], eax
-           setc al
-    };
 }
 #pragma warning(pop)
 #endif	/* _M_CEE_PURE */
@@ -3917,7 +3878,7 @@
 #else
 
 #if !defined(_M_CEE_PURE)
-#define YieldProcessor() __asm { rep nop }
+#define YieldProcessor()
 #endif  // !defined(_M_CEE_PURE)
 
 #endif  // (_MSC_FULL_VER >= 140041204)
@@ -3940,10 +3901,6 @@
     VOID
     )
 {
-    LONG Barrier;
-    __asm {
-        xchg Barrier, eax
-    }
 }
 #pragma warning( pop )
 
@@ -3987,10 +3944,6 @@
     )
 
 {
-    __asm {
-        mov ecx, Counter
-        rdpmc
-    };
 }
 
 #endif
@@ -4019,7 +3972,6 @@
     )
 
 {
-    __asm rdtsc
 }
 
 #endif
@@ -4061,7 +4013,6 @@
     )
 
 {
-    __asm int 0x2c
 }
 
 #pragma warning( pop )
@@ -4082,12 +4033,9 @@
 #pragma warning(push)
 #endif
 #pragma warning (disable:4035 4793)        // disable 4035 (function must return something)
-__inline PVOID GetFiberData( void ) { __asm {
-                                        mov eax, fs:[0x10]
-                                        mov eax,[eax]
-                                        }
+__inline PVOID GetFiberData( void ) {
                                      }
-__inline PVOID GetCurrentFiber( void ) { __asm mov eax, fs:[0x10] }
+__inline PVOID GetCurrentFiber( void ) {  }
 
 #if _MSC_VER >= 1200
 #pragma warning(pop)
@@ -12958,7 +12906,6 @@
     )
 
 {
-    __assume_bound(TagBase);
     return ((DWORD)((TagBase) + ((Tag) << HEAP_TAG_SHIFT)));
 }
 #endif
@@ -14616,7 +14563,7 @@
 } ENLISTMENT_INFORMATION_CLASS;
 
 typedef struct _TRANSACTION_LIST_ENTRY {
-    UOW    UOW;
+    ::UOW    UOW;
 } TRANSACTION_LIST_ENTRY, *PTRANSACTION_LIST_ENTRY;
 
 typedef struct _TRANSACTION_LIST_INFORMATION {
@@ -14989,7 +14936,7 @@
 
 #pragma warning (disable:4035)        // disable 4035 (function must return something)
 
-__inline struct _TEB * NtCurrentTeb( void ) { __asm mov eax, fs:[PcTeb] }
+__inline struct _TEB * NtCurrentTeb( void ) { }
 
 #if _MSC_VER >= 1200
 #pragma warning(pop)
